common/build/eslint-config-fluid/src/custom-rules/no-restricted-tags-imports.js

```diff
@@ -0,0 +1,161 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+const path = require("path");
+const { Project } = require("ts-morph");
+
+/**
+ * Validates if a tag is correctly formatted.
+ * @param tag - The tag to be validated.
+ * @param context - The context object provided by ESLint.
+ * @returns true if valid tag, else false.
+ */
+function isTagValid(tag, context) {
+	if (!tag.startsWith("@")) {
+		context.report({
+			loc: { line: 1, column: 0 },
+			message: `Invalid tag format in rule configuration: '{${tag}}'. Tags should start with '@'.`,
+			data: { tag },
+		});
+	} else {
+		return true;
+	}
+}
+
+/**
+ * Processes the tags array to ensure all tags are valid.
```

Comment:
Nit: Might be a bit more accurate to say "filters the tags" and note that we will report any invalid ones we get.